lib/client/ipam_block_reader_writer.go

```diff
@@ -82,10 +82,9 @@ func (rw blockReaderWriter) claimNewAffineBlock(
 		for _, pool := range givenPools {
 			// Validate the given pool is actually configured and matches the version.
 			if err := rw.checkPoolAvailability(pool); err != nil {
```

Comment:
im going to refactor that code, not sure why it's done twice, we could just make one call and then check for the ones we care about